File: cpp/1701.average-waiting-time.cpp
#include <vector>

using std::vector;

class Solution {
public:
  double averageWaitingTime(vector<vector<int>> &customers) {
    long total = 0, start = 0, idleAt = 0;

    for (const auto &c : customers) {
      start = idleAt > c[0] ? idleAt : c[0];
      total += start - c[0];
      total += c[1];
      idleAt = start + c[1];
    }

    return (double)total / customers.size();
  }
};

File: python/1701.average-waiting-time.py


class Solution:
    def averageWaitingTime(self, customers: list[list[int]]) -> float:
        n = len(customers)
        totalWait = 0
        idleAt = 0

        for i in range(n):
            start = max(customers[i][0], idleAt)
            totalWait += start - customers[i][0]
            totalWait += customers[i][1]
            idleAt = start + customers[i][1]

        return totalWait / n

